### ISSUE TITLE
Unify the Docker networking configuration

### ISSUE THREAD

**Issue by m09526** [gchq/Palisade#221]
In the Docker Multi JVM example, the client runs outside of a docker container while all the servers run inside docker containers. This makes networking odd, since the IP address (and ports) for container <-> container networking are different from client <-> container networking. This means that in some places the services must be aware of whether they are running inside containers or not. This means that currently the use of containers is a leaky abstraction, since it shouldn't matter to them. As we implement service auto discovery this becomes a blocker since the IP address reported by services in containers will not be usable outside of the container to access that service.

Change the docker example so that the client runs inside a container as well. This will then enable the removal of a lot of "containerIP" code inside ExampleConfigurator as well.

**Comment by ac74475** [comment]
I agree that we should containerise the client as well. A lot of the ExampleConfigurator stuff is getting changed within gh-214 so it is probably best to wait until that is merged before starting this issue

**Comment by ac74475** [comment]
Turns out sorting gh-214 is really difficult without sorting this first
